docs/src/architecture/state.md

```diff
@@ -1,74 +1,86 @@
+Nullifiers are stored in a [Tiered Sparse Merkle Tree](https://0xpolygonmiden.github.io/miden-base/crypto-primitives/tsmt.html), which maps [Note Nullifiers](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-nullifier) to `0` or `1`. Nullifiers provide information on whether a specific note has been consumed yet. The database allows proving that a given nullifier is not in the database.
 
 <p align="center">
   <img src="../diagrams/architecture/state/Nullifier_DB.png">
 </p>
 
-To prove that nullifier `11` is not in the database the operator needs to provide a Merkle path to its node, and then show that the value in that node is $0$. In our case nullifiers are 32 bytes each, and thus, the height of the Sparse Merkle tree need to be 256.
+To prove that a certain note is not yet consumed, the operator needs to provide a Merkle path to its node, and then show that the value in that node is `0`. In our case nullifiers are $32$ bytes each, and thus, the height of the Sparse Merkle tree need to be $256$.
 
 To be able to add new nullifiers to the database, operators needs to maintain the entire nullifier set - otherwise they would not be able to compute the new root of the tree.
 
-There will be one tree per epoch (~3 months), and Miden nodes always store trees for at least two epochs. However, the roots of the old trees are still stored. If a user wants to consume a note that is more than 6 month old, there must be a merkle path provided to the Miden Node for verification.
+*Note: Nullifiers as constructed in Miden break linkability of privately stored notes and the information about the note's consumption. To know the [note's nullifier](https://0xpolygonmiden.github.io/miden-base/architecture/notes.html#note-nullifier) one must know the note's data. 
+
+There will be one tree per epoch (~3 months), and Miden nodes always store trees for at least two epochs. However, the roots of the old trees are still stored. If a user wants to consume a note that is more than $6$ month old, there must be a merkle path provided to the Miden Node for verification.
+
+## State bloat minimization
+Operators don’t need to know the entire state to verify or produce a new block. No operator is required to store the entire state.
+
+At its core, the idea is simple: Instead of storing the full state data with the operators, the users store their data, and the rollup only keeps track of commitments to the data. At least for most accounts, some smart contracts need to be publicly visible. This minimizes state bloat—as the operator doesn’t need to store an ever-growing database—and provides privacy because all other users and the operator only see a hash of other users’ data.
```

Comment:
At least for most accounts, some smart contracts need to be publicly visible. -> At least for private accounts, some smart contracts need to be publicly visible.

I don't think we should make assumptions about how the network should be used in terms of private / public accounts.

Reply:
makes sense